numpyro/infer/elbo.py

```diff
@@ -311,53 +312,115 @@ class RenyiELBO(ELBO):
+        of those plates will be the same across particles.
 
     **References:**
 
     1. *Renyi Divergence Variational Inference*, Yingzhen Li, Richard E. Turner
     2. *Importance Weighted Autoencoders*, Yuri Burda, Roger Grosse, Ruslan Salakhutdinov
     """
 
-    def __init__(self, alpha=0, num_particles=2):
+    def __init__(self, alpha=0, num_particles=2, create_plates=None):
         if alpha == 1:
             raise ValueError(
                 "The order alpha should not be equal to 1. Please use ELBO class"
                 "for the case alpha = 1."
             )
         self.alpha = alpha
+        self.create_plates = create_plates
         super().__init__(num_particles=num_particles)
 
+    def _create_plates(self, key, *args, **kwargs):
+        with seed(rng_seed=key):
+            plates = self.create_plates(*args, **kwargs)
+        if isinstance(plates, plate):
+            plates = [plates]
+        assert all(
+            isinstance(p, plate) for p in plates
+        ), "create_plates() returned a non-plate"
+        plate_values = {}
+        for p in plates:
+            with p as idx:
+                plate_values[p.name] = idx
+        return plate_values
+
     def loss(self, rng_key, param_map, model, guide, *args, **kwargs):
-        def single_particle_elbo(rng_key):
+        def single_particle_elbo(rng_key, indep_subsample_plates=()):
             model_seed, guide_seed = random.split(rng_key)
             seeded_model = seed(model, model_seed)
             seeded_guide = seed(guide, guide_seed)
-            guide_log_density, guide_trace = log_density(
-                seeded_guide, args, kwargs, param_map
-            )
-            # NB: we only want to substitute params not available in guide_trace
-            model_param_map = {
-                k: v for k, v in param_map.items() if k not in guide_trace
-            }
-            seeded_model = replay(seeded_model, guide_trace)
-            model_log_density, model_trace = log_density(
-                seeded_model, args, kwargs, model_param_map
+            model_trace, guide_trace = get_importance_trace(
+                seeded_model, seeded_guide, args, kwargs, param_map
             )
             check_model_guide_match(model_trace, guide_trace)
             _validate_model(model_trace, plate_warning="loose")
 
+            site_plates = {
+                name: {frame for frame in site["cond_indep_stack"]}
+                for name, site in model_trace.items()
+                if site["type"] == "sample"
+            }
+            if site_plates:
+                common_plates = set.intersection(*site_plates.values())
+            else:
+                common_plates = set()
+            common_plate_scale = 1.0
+            for frame in common_plates:
+                subsample_size = frame.size
+                size = model_trace[frame.name]["args"][0]
+                if size > subsample_size and frame.name not in indep_subsample_plates:
```

Comment:
what is the difference between `frame`s that are in and not in `indep_subsample_plates`?

Reply:
I just added comments in the code. Basically this decides whether or not we want to keep those subsample plate dimensions in the important weight.